src/sources/mongodb_metrics/types.rs

```diff
@@ -0,0 +1,431 @@
+pub struct CommandServerStatusMetricsQueryExecutor {
+    pub scanned: i64,
+    pub scanned_objects: i64,
+    pub collection_scans: Option<CommandServerStatusMetricsQueryExecutorCollections>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CommandServerStatusMetricsQueryExecutorCollections {
+    pub total: i64,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CommandServerStatusMetricsRecord {
+    pub moves: i64,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CommandServerStatusMetricsRepl {
+    pub apply: CommandServerStatusMetricsReplApply,
+    pub buffer: CommandServerStatusMetricsReplBuffer,
+    pub executor: CommandServerStatusMetricsReplExecutor,
+    pub network: CommandServerStatusMetricsReplNetwork,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CommandServerStatusMetricsReplApply {
+    pub batches: CommandServerStatusMetricsReplApplyBatches,
+    pub ops: i64,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusMetricsReplApplyBatches {
+    pub num: i32,
+    pub total_millis: i32,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusMetricsReplBuffer {
+    pub count: i64,
+    pub max_size_bytes: i64,
+    pub size_bytes: i64,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusMetricsReplExecutor {
+    pub queues: CommandServerStatusMetricsReplExecutorQueues,
+    pub unsignaled_events: i32,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusMetricsReplExecutorQueues {
+    pub network_in_progress: i32,
+    pub sleepers: i32,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusMetricsReplNetwork {
+    pub bytes: i64,
+    pub getmores: CommandServerStatusMetricsReplNetworkGetmores,
+    pub ops: i64,
+    pub readers_created: i64,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusMetricsReplNetworkGetmores {
+    pub num: i32,
+    pub total_millis: i32,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusMetricsReplTtl {
+    pub deleted_documents: i64,
+    pub passes: i64,
+}
+
+/// https://docs.mongodb.com/manual/reference/operator/aggregation/collStats/#latency-stats-document
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusOpLatenciesStat {
+    pub latency: i64,
+    pub ops: i64,
+    pub histogram: Vec<CommandServerStatusOpLatenciesStatHistBucket>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CommandServerStatusOpLatenciesStatHistBucket {
+    pub micros: i64,
+    pub count: i64,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CommandServerStatusStorageEngine {
+    pub name: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandServerStatusWiredTiger {
```

Comment:
I wonder if we could have had some sort of strategy to isolate the metrics in https://github.com/timberio/vector/pull/4500/files#r506717666 grouped up here so it was easier to audit?